src/logics/ArithLogic.h

```diff
@@ -222,10 +229,10 @@ class ArithLogic : public Logic {
 
     bool isNumVar(SymRef sr) const { return isVar(sr) and (yieldsSortInt(sr) or yieldsSortReal(sr)); }
     bool isNumVar(PTRef tr) const { return isNumVar(getPterm(tr).symb()); }
-    bool isNumVarLike(SymRef sr) const {
-        return yieldsSortNum(sr) and not isPlus(sr) and not isTimes(sr) and not isNumConst(sr);
+    bool isMonomial(SymRef sr) const {
```

Comment:
Do we need to have a method that takes a symbol?
I think we should have it only for `PTRef`.

Reply:
We have such methods for all other operations though...
Both `SymRef` and `PTRef`, for example `isNumVar`

But Ok, can remove

Reply:
There is also a check with `SymRef` at the line 557
I would leave it for now, we have two functions for everything in Arith

Reply:
We have such methods where it makes sense, but here being a monomial is not a property of the symbol, it is a property of the `Pterm`.
You can easily fix the place you mention by passing `PTRef` instead of `SymRef`.

Reply:
Removed